### PR TITLE
Introduce `CellIDTranslator.translate_level()`

### PR DESCRIPTION
**Pull request by peterrum**
sometimes one needs a unique index within a level

### CODE REVIEW

File: include/deal.II/grid/cell_id_translator.h [hunk 1]
```diff
@@ -54,6 +54,11 @@ namespace internal
   class CellIDTranslator
   {
   public:
+    /**
+     * Default constructor.
+     */
+    CellIDTranslator();
```
Comment (peterrum) [inline include/deal.II/grid/cell_id_translator.h:60]: For computation of the level index, one does not need to specify the number coarse-cells and refinements.

File: include/deal.II/grid/cell_id_translator.h [hunk 2]
```diff
@@ -187,16 +211,27 @@ namespace internal
                     dim == Accessor::structure_dimension,
                   "The information can only be queried for cells.");
 
-    types::global_cell_index id = 0;
+    Assert(!tree_sizes.empty(), ExcNotImplemented());
```
Comment (peterrum) [inline include/deal.II/grid/cell_id_translator.h:214]: I added some asserts, in the case that someone calls these function with the new default constructor.

**Comment by drwells** [comment]
Would you be willing to make this a static function instead of adding a default ctor? That would make it simpler to call.

**Comment by peterrum** [comment]
@drwells Done!

**Review by drwells — APPROVED** [review]

**Pull request merged**